internal/api/errors.go

```diff
@@ -96,6 +96,10 @@ func conflictError(fmtString string, args ...interface{}) *HTTPError {
 	return httpError(http.StatusConflict, ErrorCodeConflict, fmtString, args...)
 }
 
+func gatewayTimeoutError(errorCode ErrorCode, fmtString string, args ...interface{}) *HTTPError {
```

Comment:
Can we avoid using this error. GoTrue is not a proxy, it's a client with respect to the HTTP hook. It should be a 422 Unprocessable Entity with a right error code that says the request to the HTTP hook timed out.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/504

Reply:
Sure 422 sounds great